sql/core/src/main/scala/org/apache/spark/sql/execution/DataSourceScanExec.scala

```diff
@@ -225,13 +225,16 @@ case class FileSourceScanExec(
   }
 
   // These metadata values make scan plans uniquely identifiable for equality checking.
-  override val metadata: Map[String, String] = Map(
-    "Format" -> relation.fileFormat.toString,
-    "ReadSchema" -> outputSchema.catalogString,
-    "Batched" -> supportsBatch.toString,
-    "PartitionFilters" -> partitionFilters.mkString("[", ", ", "]"),
-    "PushedFilters" -> dataFilters.mkString("[", ", ", "]"),
-    "InputPaths" -> relation.location.paths.mkString(", "))
+  override val metadata: Map[String, String] = {
+    def seqToString(seq: Seq[Any]) = seq.mkString("[", ", ", "]")
+    val location = relation.location
+    Map("Format" -> relation.fileFormat.toString,
+      "ReadSchema" -> outputSchema.catalogString,
+      "Batched" -> supportsBatch.toString,
+      "PartitionFilters" -> seqToString(partitionFilters),
+      "PushedFilters" -> seqToString(dataFilters),
+      "Location" -> (location.getClass.getSimpleName + seqToString(location.rootPaths)))
```

Comment:
Is it different from location.getRootPaths.length? I guess that would only be meaningful for Hive-backed tables, but that seems ok.

Btw, you might want to take(10) on the root paths to avoid materializing a large string here before it gets truncated.